src/arena.c

```diff
@@ -2054,9 +2054,11 @@ arena_new(tsdn_t *tsdn, unsigned ind, extent_hooks_t *extent_hooks) {
 		/* background_thread_create() handles reentrancy internally. */
 		if (have_background_thread) {
 			bool err;
+			malloc_mutex_unlock(tsdn, &arenas_lock);
```

Comment:
I checked through the code path however it didn't seem quite possible. We need to protect ```arena_choose_hard()``` which walks all arenas to make the decision. How about I move the ```background_thread_create()``` around so that it's outside of arenas_lock's scope?